parsing/parser.mly

```diff
@@ -2388,8 +2381,10 @@ val_longident:
 ;
 constr_longident:
     mod_longident       %prec below_DOT         { $1 }
+  | mod_longident DOT LPAREN COLONCOLON RPAREN  { Ldot($1,"::") }
```

Comment:
They are currently interpreted as delimited local open, in other words they are mapped to `(let open M in ())` or `(let open M in [])`, which means that
```OCaml 
    module M = struct end
    let ( ) = M.( )
```
is well typed but raises a warning(33) for unused open.